instagram-app/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import PostContainer from "./PostContainer";
+import SearchBar from "../SearchBar/SearchBar";
+import dummyData from "../../dummy-data";
+
+class PostsPage extends React.Component {
+	constructor() {
+		super();
+		this.state = {
+			data: [],
+			searchValue: ""
+		};
+	}
+
+	componentDidMount() {
+		this.setState({
+			data: dummyData
+		});
+	}
+
+	searchChange = e => {
+		this.setState({
+			searchValue: e.target.value
+		});
+
+		const searchTerm = e.target.value;
+		this.state.data.map(user => {
```

Comment:
I should have noticed this yesterday. We can't manipulate state directly in react, check out this https://daveceddia.com/why-not-modify-react-state-directly/ or this https://stackoverflow.com/questions/37755997/why-cant-i-directly-modify-a-components-state-really , and I would do this instead
```js
search = e => {
    const searchTerm = e.target.value;
    this.setState(prevState => ({
      userData: prevState.userData.map(user => {
        if (user.username.includes(searchTerm)) {
          user.display = "";
          return user;
        } else {
          user.display = "no";
          return user;
        }
      })
    }));
  };
```